tools/GSDumpGUI/Forms/frmMain.cs

```diff
@@ -107,15 +116,61 @@ public GSDumpGUI()
 
             _availableGsDumps.OnIndexUpdatedEvent += UpdatePreviewImage;
 
-            txtGSDXDirectory.DataBindings.Add(nameof(TextBox.Text), Settings, nameof(Settings.GSDXDir));
-            txtDumpsDirectory.DataBindings.Add(nameof(TextBox.Text), Settings, nameof(Settings.DumpDir));
+            this.Text += " " + (IntPtr.Size * 8).ToString() + "bits";
+
+            if (String.IsNullOrEmpty(Settings.GSDXDir) || !Directory.Exists(Settings.GSDXDir))
+                Settings.GSDXDir = AppDomain.CurrentDomain.BaseDirectory;
+
+            if (String.IsNullOrEmpty(Settings.DumpDir) || !Directory.Exists(Settings.DumpDir))
+                Settings.DumpDir = AppDomain.CurrentDomain.BaseDirectory;
+
+            txtGSDXDirectory.Text = Settings.GSDXDir;
+            txtDumpsDirectory.Text = Settings.DumpDir;
 
             BindListControl(lstDumps, _availableGsDumps, g => g.Files, f => f.DisplayText, g => g.SelectedFileIndex);
             BindListControl(lstGSDX, _availableGsDlls, g => g.Files, f => f.DisplayText, g => g.SelectedFileIndex);
 
             Processes = new List<Process>();
 
             NoImage = CreateDefaultImage();
+
+            _dllWatcher = new List<FileSystemWatcher>();
+            _dumpWatcher = new List<FileSystemWatcher>();
+            _watcherEvents = new ConcurrentQueue<int>();
+
+            Thread watcherThread = new Thread(ChangesWatchdogThread);
```

Comment:
There is also something like a BackgroundThread for such work but might not be available on this framework version

Reply:
`System.ComponentModel.BackgroundWorker` it's apparently available, but it seems to be to fire a single task in the background and be done with it.

I made the thread because the filewatchers would spam with reloads. The thread kinda works as a way to reduce the spam and just run one reload per many requests. So it's the message consumer thread for the multiple producers. Would BackgroundWorker work for this?

Reply:
It’s basically the same as thread but with lower cpu prio and some special long running thread pool settings iirc. But a timer would be better.